client/hwlib/src/utils.rs

```diff
@@ -0,0 +1,74 @@
+/* Copyright 2024 Canonical Ltd.
+ * All rights reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Written by:
+ *        Nadzeya Hutsko <nadzeya.hutsko@canonical.com>
+ */
+
+use once_cell::sync::Lazy;
+use std::fs::read_dir;
+use std::io::ErrorKind;
+use std::path::PathBuf;
+use std::{env, io};
+
+pub fn get_test_filepath(file_name: &str) -> &'static str {
+    fn build_test_filepath(file_name: &str) -> String {
+        let mut path = get_project_root().unwrap();
+        if !path.to_str().unwrap().contains("hwlib") {
+            // If hwlib is not in the path, the project root is the monorepo root
+            path.push("client");
+            path.push("hwlib");
+        }
+        path.push("tests");
+        path.push("test_data");
+        path.push(file_name);
+        path.to_str().unwrap().to_string()
+    }
+
+    static TEST_ENTRY_FILE: Lazy<String> = Lazy::new(|| build_test_filepath("smbios_entry_point"));
```

Comment:
Is the reason for requiring `Lazy` here that you need to return static strings to satisfy the API used in `create_certification_status_request`? If so it's probably a lot easier to change those fields to `Strings` instead.

Reply:
The problem with using strings is that for default values we use constants, and AFAIK, you cannot use `String` for constaints

Reply:
Not as the constant itself, no. But if you make the switch to using the struct approach I outlined above instead you can make those fields `Strings` and then everything is just a `String` :slightly_smiling_face:.

Reply:
I'm using the struct approach, but the values in the struct are `&'static str`. Also, we use the default values for smbios files from the `smbioslib`, and they are constants with of same type
Probably I misunderstand something, should I convert the `&'static str` to `String`?

Reply:
Sorry that wasn't very clear, this would involve the strict having string values, yeah 👍